trieste/utils/objectives.py

```diff
@@ -164,6 +164,85 @@ def hartmann_3(x: TensorType) -> TensorType:
 """
 
 
+def rosenbrock_4(x: TensorType) -> TensorType:
+    """
+    The Rosenbrock function, rescaled to have zero mean and unit variance over :math:`[0, 1]^4. See
+    :cite:`Picheny2013` for details.
+
+    This function (also known as the Banana function) is unimodal, however the minima
+    lies in a narrow valley.
+
+    :param x: The points at which to evaluate the function, with shape [..., 4].
+    :return: The function values at ``x``, with shape [..., 1].
+    :raise ValueError (or InvalidArgumentError): If ``x`` has an invalid shape.
+    """
+    tf.debugging.assert_shapes([(x, (..., 4))])
+
+    x = 15.0 * x - 5
```

Comment:
For some reason the type checker has decided that after this statement, x is guaranteed to be a basic Python float, which it obviously isn't. I think this might actually a bug with mypy (I've found a minimal example).

In the meantime, you can either use a different variable with an explicit type annotation:

```suggestion
    y : TensorType = 15.0 * x - 5
```

Or swap the mulitplicands around:

```suggestion
    x = x * 15.0 - 5
```

As an aside, you should perhaps add a comment somewhere about the scaling?